examples/Kindle2Folder.rb

```diff
@@ -0,0 +1,29 @@
+require 'kindle_highlights'
+
+def sanitize_filename(filename) #Function to remove invalid symbols from filenames
+  fn = filename.split /(?<=.)\.(?=[^.])(?!.*\
+  fn.map! { |s| s.gsub /[^a-z0-9\-]+/i, '_' }
+  return fn.join '.'
+end
+
+kindle = KindleHighlights::Client.new(
+  email_address: "aaa@aaa.com",
+  password: "password"
+  )
+
+folder = File.expand_path('~/Kindle') #Folder you want to store the txt files (in markdown format) in...
+
+kindle.books.each do |book|
+  fname = folder+"/"+sanitize_filename(book.title)+".txt"
```

Comment:
The Ruby convention is to prefer String interpolation to concatenation (see https://relishapp.com/womply/ruby-style-guide/docs/strings, among others). I'd also strongly encourage you to name your variables a bit more descriptively.

```ruby
folder_name = "#{folder}/#{sanitize_filename(book.title)}.txt"
```